examples/hubert/lightning.py

```diff
@@ -0,0 +1,139 @@
+from typing import Tuple
+
+import torch
+import torchaudio
+from dataset import BucketizeSampler, DistributedBatchSampler, HuBERTDataSet, CollateFnHubert
+from loss import hubert_loss
+from pytorch_lightning import LightningModule
+from torch import Tensor
+from torch.optim.optimizer import Optimizer
+from torch.utils.data import DataLoader
+
+
+Batch = Tuple[Tensor, Tensor, Tensor]
+
+
+class PolynomialDecayLRScheduler(torch.optim.lr_scheduler._LRScheduler):
+    """PolynomialDecay learning rate scheduler with warm up."""
+
+    def __init__(
+        self,
+        optimizer: Optimizer,
+        warmup_updates: int,
+        max_updates: int,
+        last_epoch: int = -1,
+        verbose: bool = False,
+    ):
+        self.warmup_updates = warmup_updates
+        self.warmup_factor = 1.0 / self.warmup_updates
+        self.max_updates = max_updates
+        super().__init__(optimizer, last_epoch=last_epoch, verbose=verbose)
+
+    def get_lr(self):
+        if self._step_count <= self.warmup_updates:
+            return [self.warmup_factor * base_lr for base_lr in self.base_lrs]
```

Comment:
in that case, should this line be amended? e.g.

```suggestion
            return [base_lr * self._step_count / self.warmup_updates for base_lr in self.base_lrs]
```